### DISCUSSION TITLE
Any example to use Audit.NET with IEndpointRouteBuilder (.NET 8)?

### DISCUSSION THREAD

**Discussion by thangchung** [thepirat000/Audit.NET#643] [Q&A]
Let's say we have an endpoint (.NET 8) like

```csharp
public static IEndpointRouteBuilder MapOrderFulfillmentApiRoutes(this IEndpointRouteBuilder builder)
{
    builder.MapGet("/v1/api/fulfillment-orders", async (ISender sender) => await sender.Send(new OrderFulfillmentQuery()));
    return builder;
}
```

Now, we want to do an audit action on the endpoint above. I searched around on the internet and this repo, but couldn't find any example to address it. Do we have any place to mention it? Thank you very much.

**Comment by thepirat000** [comment]
Yes, you could use the [Audit Middleware](https://github.com/thepirat000/Audit.NET/tree/master/src/Audit.WebApi#3--middleware)

For example, in your program.cs:

```c#
var app = builder.Build();

// Add and configure the audit middleware
app.UseAuditMiddleware(config => config
    .IncludeHeaders()
    .IncludeResponseHeaders()
    .FilterByRequest(r => r.Path.Value.StartsWith("/v1/api/fulfillment-orders")));
```

**Reply by thangchung** [reply]
Thanks @thepirat000, I will check it
